server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -1496,7 +1523,7 @@ public void flush(FlushRequest request) {
      * {@link org.opensearch.index.translog.TranslogDeletionPolicy} for details
      */
     public void trimTranslog() {
-        if (isRemoteTranslogEnabled()) {
+        if (isRemoteTranslogEnabled() || isMigratingToRemote()) {
```

Comment:
Since migration is dual mode, shouldn't we allow the trim here?

Reply:
The intent is to keep the behavior same as remote translog today.